yt_idefix/data_structures.py

```diff
@@ -759,3 +788,110 @@ def _is_valid(cls, filename: str, *args, **kwargs) -> bool:
 
     def _get_log_file(self) -> str:
         return os.path.join(self.directory, "vtk.out")
+
+
+class PlutoXdmfDataset(StaticPlutoDataset):
+    _dataset_type = "pluto-xdmf"
+    _index_class = PlutoXdmfHierarchy
+    _field_info_class = PlutoXdmfFields
+
+    def __init__(
+        self,
+        filename,
+        *,
+        eos_params: dict[str, float] | None = None,
+        **kwargs,
+    ):
+        super().__init__(filename, **kwargs)
+        self.parameters["equation_of_state"] = eos_params or {}
```

Comment:
@dutta-alankar  this part should be removed too

Reply:
Let's keep the `__init__()` function but remove the `equation_of_state` line. This will allow easier incorporation of this or some other features in future by passing to `yt.load()`